### DISCUSSION TITLE
Хотелки по поводу движка

### DISCUSSION THREAD

**Discussion by KillMurphy** [OGSR/OGSR-Engine#285] [Ideas]
1. В ЧН и ЗП ночью на заднем плане играет зацикленный звук сверчка
sounds\ambient\background\night_bkg_1.ogg

А в подземках играет underground_bkg_1.ogg

Можно ли такую фичу добавить в OGSR?

2. Хотел попросить добавить поддержку ЗП систему смены визуала рук, но оказывается, что три месяца назад кто-то уже просил это сделать. Просто руки ТЧ очень неудобны на самом деле + ограничение костей в 64 (вроде как в OGSR лимит расширили, но SDK всё равно не хочет экспортировать модель из-за превышения костей в 64)

3. В Prosectors Project есть такая фича, что пули вылетают из дула ствола, а не центра экрана как в оригинале. Мне лично такое очень понравилось и как я понял, у них реализовано через двигло

**Comment by xrSimpAI** [comment]
1. Это эмбиент погоды. К движку это не имеет отношения.
2. Здесь в движке столько всего наворочено по оружию, что адаптировать под это систему с отдельными руками + поддерживать старые модели от ТЧ никто не будет. Да и какой смысл. Если бы условный ганслингер был на исходниках, его можно было бы перенести в отдельный xrGame, т.к. там качество отличное. А всякие околованильные оружейные паки ЗП - не сказать что прям супер офигенные, ради них это делать смысла нет.
3. В Ганслингере тоже так сделано, насколько я помню. Ну можно подумать.

**Comment by Xottab-DUTY** [comment]
Пули из дула ствола легко сделать.
v2v3v4 выкладывал код либо на gameru, либо в Окси, либо и там и там. 🙂

**Comment by KillMurphy** [comment]
Насчёт эмбиента погоды. Я просто очень хочу добавить это себе в сборку и смотрю как сделано в ЗП:
[night]
        effects                          = effect_1, effect_2, effect_3, effect_8
        max_effect_period                = 160.000000
        min_effect_period                = 130.000000
        sound_channels                   = night_bkg_1, bkwind_1, bkwind_2, swampwind_1, crickets_1, crickets_2, owls_1, howling_1, dark_1

[night_bkg_1]
        max_distance                     = 31.000000
        min_distance                     = 30.000000
        period0                          = 0
        period1                          = 0
        period2                          = 1
        period3                          = 1
        sounds                           = ambient\background\night_bkg_1

А в ТЧ вот так

[ambient_env_night]
sound_period		= 5, 10					; <from,to>				(sec)
sound_dist			= 30, 60
sounds				= (имена звуковых файлов)
effect_period		= 130, 160					; <from,to>				(sec)
effects				= ae0_effect_1,ae0_effect_2, ae0_effect_3,ae0_effect_8	; <eff0,eff1..effN>		effects sect name

Поэтому зациклить определенный звук у меня не выйдет, как я понял. Или всё же можно?

**Comment by KillMurphy** [comment]
А, и ещё: что насчёт того, чтобы добавить поддержку визуала ног из Lost Alpha DC и Dead Air?

**Comment by xrLil-Batya** [comment]
> А, и ещё: что насчёт того, чтобы добавить поддержку визуала ног из Lost Alpha DC и Dead Air?

Вот, что нашел: https://bitbucket.org/sv3nk/la-code/commits/?search=Legs
Из репо ЛА

**Comment by xrSimpAI** [comment]
Не могу сказать, что мне нравится то что я там вижу.

**Comment by KillMurphy** [comment]
Хотел спросить насчёт фичи с патронником, планируете добавить в двигло или нет?

**Comment by xrSimpAI** [comment]
Лично я вообще не планирую никаких правок по оружию.

**Comment by KillMurphy** [comment]
Кстати, я как-то упоминал о проблеме с музыкальным эмбиентом. Мол, не работает с 22:00 до 24:00. А если в конфиге выставлено как-то так:

[escape_musics]
music\amb01				=  6, 22,	0.6,	20, 40
music\night				=  22, 6,	0.9,	20, 40


То до 6 утра будет гробовая тишина. Код из ЗП создал пару багов, а вот что если попробовать из Lost Alpha DC? Там вроде доведено до ума всё. Правда, нужно будет поправить конфиг для подземок, ибо в ТЧ оно выставлено вот так:

music\amb10					=  0, 0,   0.6,   20, 40

а в LA DC сделано так: 

music\amb10					=  0, 24,   0.6,   20, 40.

вот ссылка на последнюю обнову бинов для LA DC:

https://www.dropbox.com/s/kyg2898rbh609zh/LADC%20Bins.rar?dl=0&file_subpath=%2FLADC+Bins

если у вас будет желание пофиксить эту проблему, то думаю надо взять код из лост альфы

**Comment by imcrazyhoudini** [comment]
А в движке отсутствует возможность менять худ рук при смене костюма? Я думал из ОГСЕ осталось.

**Comment by xrSimpAI** [comment]
в огсе никакой смены рук быть не могло. Я добавлял смену текстур, но модели рук не меняются. Это зп-система, под нее мне лень всё переделывать, тем более годного контента нет, ради которого это можно было бы сделать.

**Comment by imcrazyhoudini** [comment]
Извиняюсь, неправильно выразился, это я и имел ввиду. Можете подсказать где в ОГСЕ можно подсмотреть эту самую смену текстур, чтобы это себе добавить?

**Comment by xrSimpAI** [comment]
**НЕ БЫЛО В ОГСЕ НИКАКОЙ СМЕНЫ ТЕКСТУР**, это мы сделали, а не огсе https://github.com/OGSR/OGSR-Resources/blob/master/gamedata/scripts/ogsr_outfit_arms.script

**Comment by KillMurphy** [comment]
> **НЕ БЫЛО В ОГСЕ НИКАКОЙ СМЕНЫ ТЕКСТУР**, это мы сделали, а не огсе https://github.com/OGSR/OGSR-Resources/blob/master/gamedata/scripts/ogsr_outfit_arms.script

Я со скриптами никогда не работал, только конфиги и модели, поэтому заранее извиняюсь за тупой вопрос:
А как сделать так, чтобы этот скрипт заработал в игре? Типа, я его скачал и закинул в папку scripts, в конфигах брони везде прописал какой arm_set юзать, а не работает. Что ещё нужно сделать?

**Comment by xrSimpAI** [comment]
Подключить к каллбекам его.
Если ты настолько элементарные вещи не понимаешь, то какой смысл что-то делать с движком?
Если ты берешь движок, ты должен быть готов и скрипты править под него и тд.
Без этих знаний хотя бы на базовом уровне, вообще нет смысла в это лезть.

**Comment by imcrazyhoudini** [comment]
> НЕ БЫЛО В ОГСЕ НИКАКОЙ СМЕНЫ ТЕКСТУР

Ну это мне сказали просто, а я цитирую всего лишь.
Спасибо.

**Comment by xrSimpAI** [comment]
Можешь передать тем кто это сказал, что они некомпетентные попугаи
